spec/unit/puppet/provider/dsc_base_provider/dsc_base_provider_spec.rb

```diff
@@ -590,6 +590,21 @@
       end
     end
 
+    context 'when DateTime is nill' do
```

Comment:
Suggest renaming this to `when handling DateTimes` and validate the following scenarios:

1. When the DateTime is a valid string
2. When the DateTime is `nil`
3. When the DateTime is an invalid string
4. When the DateTime is an invalid type (integer, hash, etc)

Reply:
The valid string check is handled above in the larger successful run checks, but can add the others.